### DISCUSSION TITLE
how to get current test context in the setup file?

### DISCUSSION THREAD

**Discussion by JieamI** [vitest-dev/vitest#3007] [Q&A] [answered]
Normally, with multi-threads enabled, vitest will start a worker for each test file. Is it possible to know the absolute path of current test file at the runtime of worker?

**Comment by sheremet-va** [comment, marked as answer]
I think `expect.getState` has the path

**Reply by JieamI** [reply]
After trying I found that `filePath` is always undefined unless you call `expect.getState` in the `it` block. However, I wish to get the filePath at the stage of runing setup file.

**Reply by 7rulnik** [reply]
Just checked. It's available as a getter in `expect.getState().testPath` in vitest 3.2.4. I guess it should be closed.
